src/Nav.js

```diff
@@ -1,16 +1,38 @@
 import React, { cloneElement } from 'react';
+import ReactDOM from 'react-dom';
 import classNames from 'classnames';
 import all from 'react-prop-types/lib/all';
+import warning from 'warning';
 import deprecated from 'react-prop-types/lib/deprecated';
+import keycode from 'keycode';
 import tbsUtils, { bsStyles, bsClass as _bsClass } from './utils/bootstrapUtils';
+import { nextEnabled, TAB, PANE } from './utils/tabUtils';
 
 import ValidComponentChildren from './utils/ValidComponentChildren';
-import createChainedFunction from './utils/createChainedFunction';
+import chain from './utils/createChainedFunction';
 
 import Collapse from './Collapse';
 
 class Nav extends React.Component {
 
+  componentDidUpdate() {
```

Comment:
I feel like a lot of this logic (and below) might be extractable into a separate `TabNav` component. Maybe expose something like `navItemProps` or `itemProps` on `Nav`, and pass down the tab props from above?